samcli/lib/sync/flows/layer_sync_flow.py

```diff
@@ -258,6 +259,30 @@ def _get_dependent_functions(self) -> List[Function]:
         return dependent_functions
 
 
+class LayerSyncFlowSkipBuildDirectory(LayerSyncFlow):
+    """
+    LayerSyncFlow special implementation that will skip build step and zip contents of CodeUri
+    """
+
+    def gather_resources(self) -> None:
+        zip_file_path = os.path.join(tempfile.gettempdir(), f"data-{uuid.uuid4().hex}")
+        self._zip_file = make_zip(zip_file_path, self._layer.codeuri)
+        LOG.debug("%sCreated artifact ZIP file: %s", self.log_prefix, self._zip_file)
+        self._local_sha = file_checksum(cast(str, self._zip_file), hashlib.sha256())
+
+
+class LayerSyncFlowSkipBuildZipFile(LayerSyncFlow):
+    """
+    LayerSyncFlow special implementation, that will skip build and upload zip file which is defined in CodeUri directly
+    """
+
+    def gather_resources(self) -> None:
+        self._zip_file = os.path.join(tempfile.gettempdir(), f"data-{uuid.uuid4().hex}")
```

Comment:
How are we cleaning up these temp dirs for this?

Reply:
It is cleaned up here in parent class: https://github.com/aws/aws-sam-cli/blob/develop/samcli/lib/sync/flows/layer_sync_flow.py#L98